tests/acceptance/standard/test_group_avatar.py

```diff
@@ -0,0 +1,292 @@
+        assert group.avatar_url is None or "gravatar" in group.avatar_url
+
+    def test__group_avatar_file_not_found_should_fail(self, group):
+        # Test handling of non-existent avatar file path - should fail
+        nonexistent_path = "/path/to/nonexistent/image.png"
+
+        config = f"""
+        projects_and_groups:
+          {group.full_path}/*:
+            group_settings:
+              avatar: "{nonexistent_path}"
+        """
+
+        # GitLabForm catches exceptions
+        with pytest.raises(SystemExit) as exc_info:
+            run_gitlabform(config, group)
+
+        # Verify it's the expected exit code for processing errors
+        assert exc_info.value.code == 2
+
+    def test__group_avatar_relative_file_not_found_should_fail(self, group):
+        # Test handling of non-existent relative avatar file path - should fail
+        original_cwd = os.getcwd()
+        temp_dir = tempfile.mkdtemp()
+
+        try:
+            os.chdir(temp_dir)
+
+            # Use non-existent relative path
+            nonexistent_relative_path = "images/nonexistent_avatar.png"
+
+            config = f"""
+            projects_and_groups:
+              {group.full_path}/*:
+                group_settings:
+                  avatar: "{nonexistent_relative_path}"
+            """
+
+            # Should fail with SystemExit since file won't be found
+            with pytest.raises(SystemExit) as exc_info:
+                run_gitlabform(config, group)
+
+            # Verify it's the expected exit code for processing errors
+            assert exc_info.value.code == 2
+
+        finally:
+            os.chdir(original_cwd)
+            shutil.rmtree(temp_dir)
+
+    def test__group_avatar_failure_with_other_settings(self, group):
+        nonexistent_path = "/path/to/nonexistent/image.png"
+        test_description = "Test description that should not be set due to avatar failure"
+
+        # Store original description to verify it doesn't change
+        original_group = group.manager.get(group.id)
+        original_description = getattr(original_group, "description", "")
+
+        config = f"""
+        projects_and_groups:
+          {group.full_path}/*:
+            group_settings:
+              description: "{test_description}"
+              avatar: "{nonexistent_path}"
+        """
+
+        # Should fail due to avatar file not found
+        with pytest.raises(SystemExit) as exc_info:
+            run_gitlabform(config, group)
+
+        # Verify it's the expected exit code for processing errors
+        assert exc_info.value.code == 2
+
+        # Refresh group data and verify description wasn't changed
+        updated_group = group.manager.get(group.id)
+        current_description = getattr(updated_group, "description", "")
+
+        # Description should remain unchanged since the process failed
+        assert current_description == original_description
+        assert current_description != test_description
```

Comment:
My understanding was that gitlabform will continue to process other settings even if avatar setup fails. So, in this case I expected gitlabform exiting with error code `2` due to avatar setup failure but the description is still configured.

Maybe I'm mistaken. Are you able to test what happens with currently released gitlabform version? Maybe an invalid config for `group_setting` but another setting (i.e. `group_hooks`)? Will group hooks still be processed if group setting fails?